integrations/weaviate/src/haystack_integrations/document_stores/weaviate/document_store.py

```diff
@@ -238,72 +233,42 @@ def _to_document(self, data: Dict[str, Any]) -> Document:
 
         return Document.from_dict(data)
 
-    def _query_paginated(self, properties: List[str], cursor=None):
-        collection_name = self._collection_settings["class"]
-        query = (
-            self._client.query.get(
-                collection_name,
-                properties,
+    def _query_paginated(self, properties: List[str]):
+
+        try:
+            result = self._collection.iterator(include_vector=True, return_properties=properties)
+        except weaviate.exceptions.WeaviateQueryError as e:
+            msg = f"Failed to query documents in Weaviate. Error: {e.message}"
+            raise DocumentStoreError(msg) from None
+        return list(result)
+
+    def _query_with_filters(self, filters: weaviate.collections.classes.filters.Filter) -> List[Dict[str, Any]]:
+
+        try:
+            # this is the default value for max number of objects to retrieve in Weaviate
+            # see QUERY_MAXIMUM_RESULTS
+            # see https://weaviate.io/developers/weaviate/config-refs/env-vars#overview
+            # and https://weaviate.io/developers/weaviate/api/graphql/additional-operators#pagination-with-offset
+            limit = 10_000
```

Comment:
Is this actually the max possibile value in Weaviate or just the default? 
Cause if it's the default it would be better to not hardcode this and use the actual value.

Reply:
It is is the default value in weaviate.

To change it the user would need to specify it when setting up their weaviate instance and the client can't query this value from the server.

Reply:
I agree.

Then should we introduce another param to the init function or an env var to control this value?

Reply:
Can't we use `limit` and `offset` similar to the previous implementation? 
We can decide on a sane default for `limit`.

Reply:
the previous `_query_with_filters` implementation did not have `limit` or `offset`

are you referring to a different function?

Reply:
Yeah, I got confused with `_query_paginated()`. My comment still stand though, can't we use pagination?

Reply:
Sure. What default values should we set for `limit` and `offset` and do you want to make it user configurable?

Reply:
I would go with a starting `limit` of `1000`, `offset` obviously at `0` and increment it by `limit` each time.

Reply:
And let's not make it configurable as of now, if people in the future ask for that to be customizable we can always change it then.